lib/ui/page/home/page/my_profile/crop_avatar/view.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright © 2022-2024 IT ENGINEERING MANAGEMENT INC,
+//                       <https://github.com/team113>
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU Affero General Public License v3.0 as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License v3.0 for
+// more details.
+//
+// You should have received a copy of the GNU Affero General Public License v3.0
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/agpl-3.0.html>.
+
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+
+import '/api/backend/schema.dart' show CropAreaInput, PointInput;
+import '/l10n/l10n.dart';
+import '/themes.dart';
+import '/ui/widget/modal_popup.dart';
+import '/ui/widget/widget_button.dart';
+import '/util/platform_utils.dart';
+import 'controller.dart';
+import 'widget/image_cropper/enums.dart';
+import 'widget/image_cropper/widget.dart';
+
+/// View for cropping avatar image.
+///
+/// Intended to be displayed with [show] method.
+class CropAvatarView extends StatelessWidget {
+  const CropAvatarView(this.image, {super.key});
+
+  /// [Image] to be cropped.
+  final Image image;
+
+  /// Displays [CropAvatarView] wrapped in [ModalPopup].
+  static Future<CropAreaInput?> show<T>(BuildContext context, Image image) {
+    return ModalPopup.show<CropAreaInput?>(
+      context: context,
+      isDismissible: false,
+      desktopPadding: const EdgeInsets.symmetric(
+        horizontal: 20,
+        vertical: 20,
+      ),
+      child: CropAvatarView(image),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = Theme.of(context).style;
+    return GetBuilder(
+      init: CropController(image: image, aspectRatio: 1),
+      builder: (controller) {
+        return Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            ImageCropper(
+              controller: controller,
+            ),
+            const SizedBox(height: 10),
+            SizedBox(
+              height: 30,
+              child: Stack(
+                alignment: Alignment.center,
+                fit: StackFit.expand,
+                children: [
+                  Positioned(
```

Comment:
@SleepySquash because the `Cancel` and `Done` button are of different lengths, the rotation buttons can't be placed exactly in the center when using a row layout.

Reply:
@pratham-sarankar, I seem to accomplish that quite easily:

```dart
Row(
  children: [
    Icon(Icons.cancel),
    Expanded(
      child: Center(
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(Icons.arrow_left),
            Icon(Icons.arrow_right),
          ],
        ),
      ),
    ),
    Icon(Icons.done),
  ],
),
```

Or am I mistaken and this is not what should be there?

Reply:
@SleepySquash

It works well with `Icon` on either side. Imagine, Text on either side with different length. 

```
Row(
  children: [
    Text("CancelButton"),
    Expanded(
      child: Center(
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(Icons.arrow_left),
            Icon(Icons.arrow_right),
          ],
        ),
      ),
    ),
    Text("CB"),
  ],
)

```

Although it's just two character difference between 'Cancel' and 'Done', I used stack to get the precision. I hope it is okay.

Reply:
@pratham-sarankar, oh, I see now. The workaround for that would be to do something like a `Stack` with two texts, one over other, just for those to be of equal width, which seems like a bad solution. And yours is a better one, yep.